src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -367,6 +397,49 @@ private void deactivateEquations(LfContingency lfContingency, EquationSystem equ
                     deactivatedEquationTerms.add(equationTerm);
                 }
             }
+
+            // deactivate equation of voltage controlled bus
+            Optional<LfBus> controlledBus = bus.getControlledBus();
+            if (bus.hasVoltageControl() && controlledBus.isPresent()) {
+                Optional<Equation> equation = equationSystem.getEquation(controlledBus.get().getNum(), EquationType.BUS_V);
+                if (equation.isPresent()) {
+                    equation.get().setActive(false);
+                    deactivatedEquations.add(equation.get());
+                }
+            }
+        }
+    }
+
+    private void removeSharedVoltageControlEquation(LfBranch branch, EquationSystem equationSystem, VariableSet variableSet,
+                                                    List<Equation> deactivatedEquations) {
+        Optional<Equation> sharedEquation = equationSystem.getEquation(branch.getNum(), EquationType.ZERO_RHO1);
+        // If present, nothing is done here as the shared control equation of the branch is removed later (with all other equations related to the branch)
+        if (!sharedEquation.isPresent() || !sharedEquation.get().isActive()) {
+            // The given branch is the first branch of the shared controllers
+            // The term related to that branch is in the shared equations of other controllers, and will be removed as such later
+
+            // Remove the shared voltage control equation on new first controller
+            DiscreteVoltageControl dvc = branch.getDiscreteVoltageControl();
+            LfBranch newFirstController = dvc.getControllers().get(0);
```

Comment:
The first controller is the controller lost during the contingency. You have to create a new shared control between the remaining controllers (the method already exists). You also have to save previous equations in order to restore the equation system after.